CHANGELOG.rst

```diff
@@ -0,0 +1,412 @@
+* FIX: If ``sampling_rate`` is ``'auto'``\ , set to first rate of ``DenseRunVariables`` (#351)
+* FIX: Get the absolute path of the test data directory (#347)
+* FIX: Update reports to be 0.7-compatible (#341)
+* ENH: Rename ``sr`` variable to more intuitive ``interval`` (#366)
+* ENH: Support ``pathlib.Path`` and other ``str``\ -castable types (#307)
+* MNT: Updates link to derivative config file in notebook (#344)
+* MNT: Add bids-validator dependency (#363)
+* MNT: Require pandas >= 0.23.0 (#348)
+* MNT: Bump grabbit version (#338)
+* CI: Ignore OSX Python 3.5 failures (#372)
+* CI: Build with Python 3.7 on Travis, deploy on 3.6 (#337)
+
+Version 0.7.0 (January 10, 2019)
+--------------------------------
+
+This is a major, API-breaking release. It introduces a large number of new features, bug fixes, and improvements.
+
+API-BREAKING CHANGES:
+
+
+* A number of entities (or keywords) have been renamed to align more closely with the BIDS specification documents:
+
+  * 'type' becomes 'suffix'
+  * 'modality' becomes 'datatype'
+  * 'acq' is removed (use 'acquisition')
+  * 'mod' becomes 'modality'
+
+* The following directories are no longer indexed by default: derivatives/, code/, stimuli/, models/, sourcedata/. They must be explicitly included using the 'include' initialization argument.
+* The grabbids module has been renamed to layout and BIDSLayout.py and BIDSvalidator.py are now layout.py and validation.py, respectively.
+* The BIDS validator is now enabled by default at layout initialization (i.e., ``validate=True``\ )
+* The ``exclude`` initialization argument has been removed.
+* ``BIDSLayout.parse_entities`` utility has been removed (use the more flexible ``parse_file_entities``\ ).
+* Calls to ``.get()`` now return ``BIDSFile`` objects, rather than namedtuples, by default (#281). The ``BIDSFile`` API has been tweaked to ensure backwards incompatibility in nearly all cases.
+* Naming conventions throughout the codebase have been updated to ensure consistency with the BIDS specs. This is most salient in the ``analysis`` module, where snake_case has been replaced with CamelCase throughout.
+
+NEW FEATURES:
+
+
+* File metadata (i.e., in JSON sidecars) is now searchable by default, and behaves just like native BIDS entities (e.g., metadata keys can be passed as arguments to ``.get()`` calls)
+* A new BIDSFile wrapper provides easy access to ``.metadata`` and ``.image``
+* HRF convolution is now supported via bundling of nistats' hemodynamic_models module; convolution is handled via the ``convolve_HRF`` transformation.
+* Named config paths that customize how projects are processed can be added at run-time (#313)
+* Preliminary support for BIDS-Derivatives RC1 (mainly core keywords)
+
+MINOR IMPROVEMENTS AND BUG FIXES:
+
+
+* Specifying 'derivatives' in a path specification now automatically includes 'bids' (#246)
+* Zenodo DOIs are now minted with new releases (#308)
```

Comment:
```suggestion
Version 0.7.0 (January 10, 2019)
--------------------------------
This is a major, API-breaking release. It introduces a large number of new features, bug fixes, and improvements.

API-BREAKING CHANGES:

* A number of entities (or keywords) have been renamed to align more closely with the BIDS specification documents:
  * 'type' becomes 'suffix'
  * 'modality' becomes 'datatype'
  * 'acq' is removed (use 'acquisition')
  * 'mod' becomes 'modality'
* The following directories are no longer indexed by default: derivatives/, code/, stimuli/, models/, sourcedata/. They must be explicitly included using the 'include' initialization argument.
* The grabbids module has been renamed to layout and BIDSLayout.py and BIDSvalidator.py are now layout.py and validation.py, respectively.
* The BIDS validator is now enabled by default at layout initialization (i.e., ``validate=True``\ )
* The ``exclude`` initialization argument has been removed.
* ``BIDSLayout.parse_entities`` utility has been removed (use the more flexible ``parse_file_entities``\ ).
* Calls to ``.get()`` now return ``BIDSFile`` objects, rather than namedtuples, by default (#281). The ``BIDSFile`` API has been tweaked to ensure backwards incompatibility in nearly all cases.
* Naming conventions throughout the codebase have been updated to ensure consistency with the BIDS specs. This is most salient in the ``analysis`` module, where snake_case has been replaced with CamelCase throughout.

NEW FEATURES:

* File metadata (i.e., in JSON sidecars) is now searchable by default, and behaves just like native BIDS entities (e.g., metadata keys can be passed as arguments to ``.get()`` calls)
* A new BIDSFile wrapper provides easy access to ``.metadata`` and ``.image``
* HRF convolution is now supported via bundling of nistats' hemodynamic_models module; convolution is handled via the ``convolve_HRF`` transformation.
* Named config paths that customize how projects are processed can be added at run-time (#313)
* Preliminary support for BIDS-Derivatives RC1 (mainly core keywords)

MINOR IMPROVEMENTS AND BUG FIXES:

* Specifying 'derivatives' in a path specification now automatically includes 'bids' (#246)
* Zenodo DOIs are now minted with new releases (#308)
```